src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SecurityController.php

```diff
@@ -0,0 +1,385 @@
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotClearCustomerSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotClearEmployeeSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotDeleteCustomerSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotDeleteEmployeeSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\SessionNotFoundException;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShop\PrestaShop\Core\Search\Filters\Security\Session\CustomerFilters;
+use PrestaShop\PrestaShop\Core\Search\Filters\Security\Session\EmployeeFilters;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class SecurityController is responsible for displaying the
+ * "Configure > Advanced parameters > Security" page.
+ */
+class SecurityController extends FrameworkBundleAdminController
+{
+    /**
+     * Show sessions listing page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @return Response
+     */
+    public function indexAction(): Response
+    {
+        $generalForm = $this->getGeneralFormHandler()->getForm();
+
+        return $this->render(
+            '@PrestaShop/Admin/Configure/AdvancedParameters/Security/index.html.twig',
+            [
+                'layoutHeaderToolbarBtn' => [],
+                'layoutTitle' => $this->trans('Security', 'Admin.Navigation.Menu'),
+                'generalForm' => $generalForm->createView(),
+            ]
+        );
+    }
+
+    /**
+     * Process the Security general configuration form.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller')) && is_granted('create', request.get('_legacy_controller')) && is_granted('delete', request.get('_legacy_controller'))"
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processGeneralFormAction(Request $request): RedirectResponse
+    {
+        return $this->processForm(
+            $request,
+            $this->getGeneralFormHandler(),
+            'General'
+        );
+    }
+
+    /**
+     * Process the Security configuration form.
+     *
+     * @param Request $request
+     * @param FormHandlerInterface $formHandler
+     * @param string $hookName
+     *
+     * @return RedirectResponse
+     */
+    protected function processForm(Request $request, FormHandlerInterface $formHandler, string $hookName): RedirectResponse
+    {
+        $this->dispatchHook(
+            'actionAdminSecurityControllerPostProcess' . $hookName . 'Before',
```

Comment:
Ooohh now I see where the mystery hook comes from, so it is indeed used

However, I don't understand the advantage of creating this protected function since it's used only once
Wouldn't it be simpler and easier to read to have this code in the controller `processGeneralFormAction` directly, this way you can use the full hook name as well

If this code was shared with another form I would understand but here it seems to complicate things for no gain

Reply:
I just read the other conversations, I understand that this will be useful later but then I suggest that the parameter contains the full hook name
This will make research in the code much easier than building them here with concatenated strings

Reply:
It's used one for the moment, but it was originally designed for Headers configuration, password configuration, htaccess rules, rewrite rules, etc.

Reply:
Yeah I got that part I understand But to make hook research easier in the code I think we could use the whole hook name This is not a super generic hook name generation (like in the Object based on the entities class name) In this case we can use the method like this

```suggestion
            $hookName,
```

```php
    return $this->processForm(
            $request,
            $this->getGeneralFormHandler(),
            'actionAdminSecurityControllerPostProcessGeneralBefore',
        );
```